src/main/webapp/js/popularitymap-loader.js

```diff
@@ -0,0 +1,54 @@
+function createMap(){
+    map = new google.maps.Map(document.getElementById('map'), {
```

Comment:
these appears to be global variables, that's dangerous because another javascript  file may be using the same variable name. 

One way is to make these variables local to the file, but keep the functions accessible from other files. to do that, try this: 

https://stackoverflow.com/questions/10644864/make-a-javascript-variable-only-avaliable-to-functions-on-the-same-js-file

Reply:
Thanks for suggestions. I tried following this answer. I'm not sure am I addressing this problem correctly?

Reply:
looks good!